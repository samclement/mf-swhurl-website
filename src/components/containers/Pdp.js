import React from 'react'
import Head from 'next/head'
import Header from '../blocks/Header'
import { getProduct } from '../../services'
import GenderHeader from '../elements/GenderHeader.js'
import styled from 'styled-components'
import ReactSwipe from 'react-swipe'

const StyledProductContainer = styled.div`
  width: 100%;
  @media only screen and (min-width: 641px) {
    display: inline-block;
    text-align: center;
  }
`

const ImageContainer = styled.div`
  width: 100%;
  @media only screen and (min-width: 641px) {
    width: 44%;
    display: inline-block;
    padding: 10px 3% 0 3%;
  }
  float: left;
`

const StyledImage = styled.img`
  width: 100%;
  margin: 0 auto;
  @media only screen and (max-width: 320px) {
    max-width: 320px;
  }
`

const ProductDetailsContainer = styled.div`
  @media only screen and (min-width: 641px) {
    width: 50%;
    display: inline-block;
  }
`

const Controls = styled.div`
  text-align: center;
  margin-bottom: 20px;
`

function Pdp(props) {
  const { name, thumbnail, designerData, priceData, sizeTaxonomyName } = props
  const mainImage = thumbnail.replace('thumbnail', 'large')
  const gender = sizeTaxonomyName.split(' ')[0].toLowerCase()
  let reactSwipeEl
  return (
    <div>
      <Head>
        <title>mf.swhurl.com | {name}</title>
      </Head>
      <Header gender={gender} />
      <GenderHeader gender={gender} />
      <StyledProductContainer>
        <ImageContainer>
          <ReactSwipe className="carousel" ref={el => (reactSwipeEl = el)}>
            {[...Array(5).keys()].map(num => {
              return (
                <div key={num}>
                  <StyledImage
                    src={mainImage.replace('_1', `_${num + 1}`)}
                    alt={name}
                  />
                </div>
              )
            })}
          </ReactSwipe>
          <noscript>
            <img src={mainImage} />
          </noscript>
          <Controls>
            <button onClick={() => reactSwipeEl.prev()}>Previous</button>
            <button onClick={() => reactSwipeEl.next()}>Next</button>
          </Controls>
        </ImageContainer>
        <ProductDetailsContainer>
          <h2 dangerouslySetInnerHTML={{ __html: designerData.name }} />
          <h3 dangerouslySetInnerHTML={{ __html: name }} />
          <p dangerouslySetInnerHTML={{ __html: priceData.formattedValue }} />
        </ProductDetailsContainer>
      </StyledProductContainer>
    </div>
  )
}

Pdp.getInitialProps = async ({ req, query }) => {
  const res = await getProduct(query, !!req)
  return res.data
}

export default Pdp
