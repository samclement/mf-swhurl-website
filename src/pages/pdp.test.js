/* eslint-env jest */

import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import 'jest-styled-components'
import '@testing-library/jest-dom/extend-expect'
import Pdp from './pdp.js'
// Nextjs routing support
import Router from 'next/router'
const mockedRouter = { push: () => {}, prefetch: () => {} }
Router.router = mockedRouter
// Snapshot diffing support
const { toMatchDiffSnapshot } = require('snapshot-diff')
expect.extend({ toMatchDiffSnapshot })
// Data
const data = require('../../test-data/pdp.json')

describe('With Snapshot Testing', () => {
  it('Plp page shows "mf.swhurl.com" H1', () => {
    const { getByText, asFragment } = render(<Pdp {...data} />)
    const firstRender = asFragment()
    expect(firstRender).toMatchSnapshot()
    fireEvent.click(getByText(/Next/i))
    expect(firstRender).toMatchDiffSnapshot(asFragment())
    fireEvent.click(getByText(/Previous/i))
    expect(firstRender).toMatchSnapshot(asFragment())
  })
  it('Plp getInitialProps', async () => {
    const res = await Pdp.getInitialProps({ query: { code: 1247661 } })
    expect(res).toMatchSnapshot()
  })
  it('Plp getInitialProps', async () => {
    const res = await Pdp.getInitialProps({ req: { query: { code: 1247661 } } })
    expect(res).toMatchSnapshot()
  })
})
