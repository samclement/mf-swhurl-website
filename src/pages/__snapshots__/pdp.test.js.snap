// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing Plp getInitialProps 1`] = `Object {}`;

exports[`With Snapshot Testing Plp getInitialProps 2`] = `Object {}`;

exports[`With Snapshot Testing Plp page shows "mf.swhurl.com" H1 1`] = `
<DocumentFragment>
  .c2 {
  color: black;
  padding-right: 5px;
}

.c2:hover {
  color: #f00;
}

.c2:visited {
  color: black;
}

.c1 {
  font-weight: bold;
  text-transform: capitalize;
  padding-right: 10px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
}

.c3 {
  width: 100%;
  text-align: center;
  text-transform: capitalize;
  margin-bottom: 0px;
  background-color: #eee;
}

.c4 {
  width: 100%;
}

.c5 {
  width: 100%;
  float: left;
}

.c6 {
  width: 100%;
  margin: 0 auto;
}

.c7 {
  text-align: center;
  margin-bottom: 20px;
}

@media only screen and (min-width:641px) {
  .c4 {
    display: inline-block;
    text-align: center;
  }
}

@media only screen and (min-width:641px) {
  .c5 {
    width: 44%;
    display: inline-block;
    padding: 10px 3% 0 3%;
  }
}

@media only screen and (max-width:320px) {
  .c6 {
    max-width: 320px;
  }
}

@media only screen and (min-width:641px) {
  .c8 {
    width: 50%;
    display: inline-block;
  }
}

<div>
    <nav
      class="c0"
    >
      <div
        class="c1"
      >
        mens
      </div>
      <a
        class="c2"
        href="/mens/shop/accessories"
      >
        Accessories
      </a>
      <a
        class="c2"
        href="/mens/shop/shoes"
      >
        Shoes
      </a>
      <a
        class="c2"
        href="/mens/shop/clothing"
      >
        Clothing
      </a>
      <a
        class="c2"
        href="/mens/shop/bags"
      >
        Bags
      </a>
      <a
        class="c2"
        href="/mens/shop/homeware"
      >
        Homeware
      </a>
    </nav>
    <nav
      class="c0"
    >
      <div
        class="c1"
      >
        womens
      </div>
      <a
        class="c2"
        href="/womens/shop/accessories"
      >
        Accessories
      </a>
      <a
        class="c2"
        href="/womens/shop/bags"
      >
        Bags
      </a>
      <a
        class="c2"
        href="/womens/shop/clothing"
      >
        Clothing
      </a>
      <a
        class="c2"
        href="/womens/shop/shoes"
      >
        Shoes
      </a>
      <a
        class="c2"
        href="/womens/shop/homeware"
      >
        Homeware
      </a>
    </nav>
    <h1
      class="c3"
    >
      Womens
    </h1>
    <div
      class="c4"
    >
      <div
        class="c5"
      >
        <div
          class="react-swipe-container carousel"
          style="overflow: hidden; visibility: visible; position: relative;"
        >
          <div
            style="overflow: hidden; position: relative; width: 0px;"
          >
            <div
              data-index="0"
              style="float: left; width: 0px; position: relative; transition-property: transform; left: 0px; transition-duration: 0ms; webkit-transition-duration: 0ms; webkit-transform: translate(0px,0)translateZ(0);"
            >
              <img
                class="c6"
                src="//assetsprx.matchesfashion.com/img/product/1247661_1_large.jpg"
              />
            </div>
            <div
              data-index="1"
              style="float: left; width: 0px; position: relative; transition-property: transform; left: 0px; transition-duration: 0ms; webkit-transition-duration: 0ms; webkit-transform: translate(0px,0)translateZ(0);"
            >
              <img
                class="c6"
                src="//assetsprx.matchesfashion.com/img/product/1247661_2_large.jpg"
              />
            </div>
            <div
              data-index="2"
              style="float: left; width: 0px; position: relative; transition-property: transform; left: 0px; transition-duration: 0ms; webkit-transition-duration: 0ms; webkit-transform: translate(0px,0)translateZ(0);"
            >
              <img
                class="c6"
                src="//assetsprx.matchesfashion.com/img/product/1247661_3_large.jpg"
              />
            </div>
            <div
              data-index="3"
              style="float: left; width: 0px; position: relative; transition-property: transform; left: 0px; transition-duration: 0ms; webkit-transition-duration: 0ms; webkit-transform: translate(0px,0)translateZ(0);"
            >
              <img
                class="c6"
                src="//assetsprx.matchesfashion.com/img/product/1247661_4_large.jpg"
              />
            </div>
            <div
              data-index="4"
              style="float: left; width: 0px; position: relative; transition-property: transform; left: 0px; transition-duration: 0ms; webkit-transition-duration: 0ms; webkit-transform: translate(0px,0)translateZ(0);"
            >
              <img
                class="c6"
                src="//assetsprx.matchesfashion.com/img/product/1247661_5_large.jpg"
              />
            </div>
          </div>
        </div>
        <div
          class="c7"
        >
          <button>
            Previous
          </button>
          <button>
            Next
          </button>
        </div>
      </div>
      <div
        class="c8"
      >
        <h2>
          Bottega Veneta
        </h2>
        <h3>
          Intrecciato leather cardholder
        </h3>
        <p>
          Â£265
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`With Snapshot Testing Plp page shows "mf.swhurl.com" H1 2`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -96,11 +96,11 @@
            <div
              style=\\"overflow: hidden; position: relative; width: 0px;\\"
            >
              <div
                data-index=\\"0\\"
-               style=\\"float: left; width: 0px; position: relative; transition-property: transform; left: 0px; transition-duration: 0ms; webkit-transition-duration: 0ms; webkit-transform: translate(0px,0)translateZ(0);\\"
+               style=\\"float: left; width: 0px; position: relative; transition-property: transform; left: 0px; transition-duration: 300ms; webkit-transition-duration: 300ms; webkit-transform: translate(NaNpx,0)translateZ(0);\\"
              >
                <img
                  class=\\"pdp__StyledImage-sc-8q6d8y-2 cnURja\\"
                  src=\\"//assetsprx.matchesfashion.com/img/product/1247661_1_large.jpg\\"
                />"
`;

exports[`With Snapshot Testing Plp page shows "mf.swhurl.com" H1 3`] = `
Object {
  Symbol(SameObject caches): Object {
    "childNodes": NodeList [
      <div>
        <nav
          class="Nav__StyledNav-sc-1jn739u-1 iVzYBs"
        >
          <div
            class="Nav__StyledTitle-sc-1jn739u-0 gsLjLB"
          >
            mens
          </div>
          <a
            class="NavLink__StyledLink-sc-1rtt1l3-0 iOdWij"
            href="/mens/shop/accessories"
          >
            Accessories
          </a>
          <a
            class="NavLink__StyledLink-sc-1rtt1l3-0 iOdWij"
            href="/mens/shop/shoes"
          >
            Shoes
          </a>
          <a
            class="NavLink__StyledLink-sc-1rtt1l3-0 iOdWij"
            href="/mens/shop/clothing"
          >
            Clothing
          </a>
          <a
            class="NavLink__StyledLink-sc-1rtt1l3-0 iOdWij"
            href="/mens/shop/bags"
          >
            Bags
          </a>
          <a
            class="NavLink__StyledLink-sc-1rtt1l3-0 iOdWij"
            href="/mens/shop/homeware"
          >
            Homeware
          </a>
        </nav>
        <nav
          class="Nav__StyledNav-sc-1jn739u-1 iVzYBs"
        >
          <div
            class="Nav__StyledTitle-sc-1jn739u-0 gsLjLB"
          >
            womens
          </div>
          <a
            class="NavLink__StyledLink-sc-1rtt1l3-0 iOdWij"
            href="/womens/shop/accessories"
          >
            Accessories
          </a>
          <a
            class="NavLink__StyledLink-sc-1rtt1l3-0 iOdWij"
            href="/womens/shop/bags"
          >
            Bags
          </a>
          <a
            class="NavLink__StyledLink-sc-1rtt1l3-0 iOdWij"
            href="/womens/shop/clothing"
          >
            Clothing
          </a>
          <a
            class="NavLink__StyledLink-sc-1rtt1l3-0 iOdWij"
            href="/womens/shop/shoes"
          >
            Shoes
          </a>
          <a
            class="NavLink__StyledLink-sc-1rtt1l3-0 iOdWij"
            href="/womens/shop/homeware"
          >
            Homeware
          </a>
        </nav>
        <h1
          class="GenderHeader__StyledH1-hihn4e-0 hBiQqN"
        >
          Womens
        </h1>
        <div
          class="pdp__StyledProductContainer-sc-8q6d8y-0 jdVYRm"
        >
          <div
            class="pdp__ImageContainer-sc-8q6d8y-1 fniJjf"
          >
            <div
              class="react-swipe-container carousel"
              style="overflow: hidden; visibility: visible; position: relative;"
            >
              <div
                style="overflow: hidden; position: relative; width: 0px;"
              >
                <div
                  data-index="0"
                  style="float: left; width: 0px; position: relative; transition-property: transform; left: 0px; transition-duration: 0ms; webkit-transition-duration: 0ms; webkit-transform: translate(0px,0)translateZ(0);"
                >
                  <img
                    class="pdp__StyledImage-sc-8q6d8y-2 cnURja"
                    src="//assetsprx.matchesfashion.com/img/product/1247661_1_large.jpg"
                  />
                </div>
                <div
                  data-index="1"
                  style="float: left; width: 0px; position: relative; transition-property: transform; left: 0px; transition-duration: 0ms; webkit-transition-duration: 0ms; webkit-transform: translate(0px,0)translateZ(0);"
                >
                  <img
                    class="pdp__StyledImage-sc-8q6d8y-2 cnURja"
                    src="//assetsprx.matchesfashion.com/img/product/1247661_2_large.jpg"
                  />
                </div>
                <div
                  data-index="2"
                  style="float: left; width: 0px; position: relative; transition-property: transform; left: 0px; transition-duration: 0ms; webkit-transition-duration: 0ms; webkit-transform: translate(0px,0)translateZ(0);"
                >
                  <img
                    class="pdp__StyledImage-sc-8q6d8y-2 cnURja"
                    src="//assetsprx.matchesfashion.com/img/product/1247661_3_large.jpg"
                  />
                </div>
                <div
                  data-index="3"
                  style="float: left; width: 0px; position: relative; transition-property: transform; left: 0px; transition-duration: 0ms; webkit-transition-duration: 0ms; webkit-transform: translate(0px,0)translateZ(0);"
                >
                  <img
                    class="pdp__StyledImage-sc-8q6d8y-2 cnURja"
                    src="//assetsprx.matchesfashion.com/img/product/1247661_4_large.jpg"
                  />
                </div>
                <div
                  data-index="4"
                  style="float: left; width: 0px; position: relative; transition-property: transform; left: 0px; transition-duration: 0ms; webkit-transition-duration: 0ms; webkit-transform: translate(0px,0)translateZ(0);"
                >
                  <img
                    class="pdp__StyledImage-sc-8q6d8y-2 cnURja"
                    src="//assetsprx.matchesfashion.com/img/product/1247661_5_large.jpg"
                  />
                </div>
              </div>
            </div>
            <div
              class="pdp__Controls-sc-8q6d8y-4 fukWF"
            >
              <button>
                Previous
              </button>
              <button>
                Next
              </button>
            </div>
          </div>
          <div
            class="pdp__ProductDetailsContainer-sc-8q6d8y-3 jydWRR"
          >
            <h2>
              Bottega Veneta
            </h2>
            <h3>
              Intrecciato leather cardholder
            </h3>
            <p>
              Â£265
            </p>
          </div>
        </div>
      </div>,
    ],
  },
}
`;
